mlflow/tracking/_model_registry/fluent.py

```diff
@@ -77,10 +77,9 @@ def register_model(
             raise e
 
     if RunsArtifactRepository.is_runs_uri(model_uri):
-        source = RunsArtifactRepository.get_underlying_uri(model_uri)
         (run_id, _) = RunsArtifactRepository.parse_runs_uri(model_uri)
         create_version_response = client.create_model_version(
-            name, source, run_id, tags=tags, await_creation_for=await_registration_for
+            name, model_uri, run_id, tags=tags, await_creation_for=await_registration_for
```

Comment:
Can we check on Databricks to make sure this doesn't break anything? I think it might break model version creation on Databricks.

Reply:
I think Databricks expects the URI to be `dbfs:/databricks/mlflow-tracking`....

Reply:
Maybe we should instead update the validation logic in MLflow server to check if the source location is within the run's artifact directory.